api/internal/geolocation.js

```diff
@@ -0,0 +1,184 @@
+/* global GeolocationCoordinates, GeolocationPosition, GeolocationPositionError */
+import ipc from '../ipc.js'
+import os from '../os.js'
+
+const isApple = os.platform() === 'darwin'
+const watchers = {}
+
+let currentWatchIdenfifier = 0
+
+class Watcher {
+  constructor (identifier) {
+    this.identifier = identifier
+    this.listener = (event) => {
+      console.log(event)
+      if (event.detail?.params?.watch?.identifier === identifier) {
+        const position = createGeolocationPosition(event.detail.params)
+        console.log(position)
+      }
+    }
```

Comment:
does it just log to console? maybe we can improve the output to make it more descriptive?

Reply:
this should not be here!